components/constellation/constellation.rs

```diff
@@ -2778,6 +2777,11 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             Some(browsing_context) => browsing_context,
         };
 
+        {
+            let session_history = self.get_joint_session_history(browsing_context.top_level_id);
+            session_history.remove_entries_for_browsing_context(browsing_context_id);
+        }
```

Comment:
Ah, this is interesting. Does this mean we're now eagerly removing entries from the session history, so the changes to `history.length` will be immediately visible?

Reply:
Yes